keps/487-kubectl-plugin/README.md

```diff
@@ -0,0 +1,545 @@
+
+```bash
+kuctl cancel taco-123
+```
+
+It might also be useful to request a cancel all, limited to the permission that the user has, a namespace, or other filter. This likely needs a confirmation.
+
+```bash
+kuctl cancel --all
+> Are you sure you want to cancel all jobs y/n?
+```
+
+Or without the prompt:
+
+```bash
+kuctl cancel --all --force
+```
+
+Or within a specific filter:
+
+```bash
+kuctl cancel --all --namespace insects
+```
+
+This command (and others) will need to be designed with multi-tenancy in mind, because the requesting user should only be interacting with the jobs they own or control.
+
+> Question for thought: is there a difference between what might be a cancel or a delete? Is there an abstraction for a group of jobs that could also be cancelled or deleted?
+
+### Watch Job
+
+After submitting a job, it's nice to be able to watch / stream logs. That should be easy to do.
+
+```bash
+kuctl watch taco-123
+```
+```console
+... makin' tacos!
+... taco 1 is cooked!
+... taco 2 is cooking.
+```
+
+### Queues
+
+I haven't used queues extensively so this likely needs to be expanded, but akin to listing jobs, we probably want to list kueues. For all queues, this could be:
+
+```bash
+kuctl queues
+```
+
+| Name | Admitted | Pending | State |
+|------|-----------|---------|------|
+| user-queue | 1 | 1 | Operational |
+
+To filter to a specific queue:
+
+```bash
+kuctl queues user-queue
+```
+
+We likely want to be able to provide more metadata, and either we could add a `describe-queue` subcommand, or have the above second example show the more verbose information. I think I like the first idea. We also likely want to expand these subcommands to include details that the original prompt warranted. I'm not familiar enough with Kueue yet to add them. This command should also support yaml/json.
+
+```bash
+kuctl queues -o json
+```
+
+
+### Future Commands
+
+In the future, if we can make this a full fledged client for interaction with jobs, we could consider the following commands.
+Note that these are not proposed to be in the first stage of this design document.
+
+#### Submit
+
+```
+# Submit, either a yaml as is
+kuctl submit jobfile.yaml
+
+# or a simpler abstraction that uses some kind of default or template
+# This would actually be really cool if we could map a community develoepd job spec (that works for other tools) into kueue
+kuctl submit <something else>
+```
+
+#### Update
+
+Is it possible to update a CRD for a kueue job? I suspect not, but actually, if it is to change the size or some policy or termination time, maybe this isn't a crazy idea.
+
+```bash
+# Note this format follows how helm sets variables
+kuctl update set path.to.attribute=thing
+
+# This does not, but it could be wanted to remove an attribute entirely (instead of trying to set the default null type)
+kuctl update remove path.to.attribute
+```
+
+### Resources
```

Comment:
This could be done, however at pod-set level. Something similar could /should be done `cluster-queue` and `cohort` , in terms of  `Total` vs `Used`.